pkg/appgateserver/server.go

```diff
@@ -51,16 +52,13 @@ type appGateServer struct {
 	// signing information holds the signing key and application address for the server
 	signingInformation *SigningInformation
 
-	// ringCache is a cache of the public keys used to create the ring for a given application
-	// they are stored in a map of application address to a slice of points on the secp256k1 curve
-	// TODO(@h5law): subscribe to on-chain events to update this cache as the ring changes over time
-	ringCache      map[string][]ringtypes.Point
-	ringCacheMutex *sync.RWMutex
+	// ringCache is used to obtain and store the ring for the application.
+	ringCache rings.RingCache
 
 	// clientCtx is the client context for the application.
 	// It is used to query for the application's account to unmarshal the supplier's account
 	// and get the public key to verify the relay response signature.
-	clientCtx sdkclient.Context
+	clientCtx relayer.QueryClientContext
```

Comment:
I think it may make sense to move `relayer.QueryClientContext` to `pkg/client/query` as well. If it's not too much trouble, I think it would make this PR easier to review if the query client refactor gets moved out to its own PR.

(Bonus points for moving `relayer.TxClientContext` as well.  :sweat_smile:)